src/Manager.js

```diff
@@ -181,85 +205,139 @@ class GiveawaysManager extends EventEmitter {
      */
     start(channel, options) {
         return new Promise(async (resolve, reject) => {
-            if (!this.ready) {
-                return reject('The manager is not ready yet.');
+            if (!this.ready) return reject('The manager is not ready yet.');
+            if (!channel?.id || !channel.isText() || channel.deleted) {
+                return reject(`channel is not a valid text based channel. (val=${channel})`);
             }
-            options.messages = (options.messages && typeof options.messages === 'object')
-                ? merge(defaultGiveawayMessages, options.messages)
-                : defaultGiveawayMessages;
-            if (!channel || !channel.id) {
-                return reject(`channel is not a valid guildchannel. (val=${channel})`);
+            if (
+                channel.isThread() && !channel.permissionsFor(this.client.user)?.has([
+                    (channel.locked || !channel.joined && channel.type === 'GUILD_PRIVATE_THREAD')
+                        ? Discord.Permissions.FLAGS.MANAGE_THREADS
+                        : Discord.Permissions.FLAGS.SEND_MESSAGES
+                ])
+            ) return reject(`The manager is unable to send messages in the provided ThreadChannel. (id=${channel.id})`);
+            if (typeof options.prize !== 'string' || options.prize.length > 256) {
+                return reject(`options.prize is not a string or longer than 256 characters. (val=${options.prize})`);
             }
-            if (!options.time || isNaN(options.time)) {
-                return reject(`options.time is not a number. (val=${options.time})`);
+            if (options.isDrop && typeof options.isDrop !== 'boolean') {
+                return reject(`options.isDrop is not a boolean. (val=${options.isDrop})`);
             }
-            if (typeof options.prize !== 'string') {
-                return reject(`options.prize is not a string. (val=${options.prize})`);
+            if (!options.isDrop && (isNaN(options.duration) || typeof options.duration !== 'number' || options.duration < 1)) {
```

Comment:
_i am writing this with a extension, thats why no multiline, sry_
would it maybe be better to set endAt to Infinity? like for pausing? cause currently it would just be NaN

Reply:
I added a default value for endAt for drop giveaways 👍